Project In Progress State

ghickman opened the pull request:
This tweaks so Project page bits so we conform to the expectations of #1379.  I've also used it to capture some other little bugs I found along the way.

todd-dembrey commented on opentech/apply/projects/templates/application_projects/project_detail.html:
```diff
@@ -166,11 +166,26 @@ <h4 class="modal__header-bar">Request Payment</h4>
                 {% endif %}
 
-                {% if contracts %}
+                {% if not object.is_in_progress and contracts %}
```
We should be showing the contract when they exist, they are relevant at all times.

ghickman replied:
Ah, I misunderstood this line in the ticket:

> Applicant can only see proposal and contracts

Fixed and rebased.

todd-dembrey commented on opentech/apply/projects/templatetags/contract_tools.py:
```diff
@@ -10,6 +10,9 @@ def user_can_upload_contract(project, user):
     if user.is_apply_staff:
         return project.status != COMMITTED
 
+    if project.status != CONTRACTING:
```
Why did you add this? If there is a contract to sign they should be able to upload, which the following logic covers.

ghickman replied:
Poor assumption on my part, we've gated access to other things after moving out of a relevant state and I thought that applied here too since the spec didn't call it out one way or another.

Removed and updated tests to reflect the correct flow.

ghickman commented on opentech/apply/projects/templates/application_projects/project_detail.html:
```diff
@@ -166,11 +166,26 @@ <h4 class="modal__header-bar">Request Payment</h4>
+                    <p><a href="#">Supporting documents</a></p>
+                    {% endif %}
+
+                    <p><a href="{{ object.contracts.first.file.url }}">Draft contract</a></p>
```
> Template updating to reflect the design (contracts subheading within supporting docs sidebar)

As per the spec (Page Variations -> In Progress -> Design (opens Page 10)) the designs show "Draft contract" in the sidebar.

Could we do a better job at finding a draft contract? In hindsight, yes, but this is here because the ticket and design say it should be, if either of those are incorrect then we should fix those.

todd-dembrey replied:
lets remove it - we've promoted contracts to its own block and are showing that at all times so the designs are no longer relevant.

ghickman replied:
Fixed in 09d602c3

ghickman commented:
> You also miss off removing the supporting documents block from the detail page when in progress.

Thanks, I must have removed the check from the model at some point!  Fixed and rebased.

todd-dembrey commented:
`{% include "application_projects/includes/supporting_documents.html" %}`
still showing here with no surrounding logic:
https://github.com/OpenTechFund/opentech.fund/blob/4048ec371c8c9c4a297da8aa7619078faae3d8a4/opentech/apply/projects/templates/application_projects/project_detail.html#L118

todd-dembrey commented:
> `{% include "application_projects/includes/supporting_documents.html" %}`
> still showing here with no surrounding logic:
> https://github.com/OpenTechFund/opentech.fund/blob/4048ec371c8c9c4a297da8aa7619078faae3d8a4/opentech/apply/projects/templates/application_projects/project_detail.html#L118

^^ still unaddressed.

ghickman commented:
> > `{% include "application_projects/includes/supporting_documents.html" %}`
> > still showing here with no surrounding logic:
> > https://github.com/OpenTechFund/opentech.fund/blob/4048ec371c8c9c4a297da8aa7619078faae3d8a4/opentech/apply/projects/templates/application_projects/project_detail.html#L118
> 
> ^^ still unaddressed.

Oops, missed that one, thanks.  Fixed in acd2b0d9
